src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -128,8 +129,38 @@
    */
   static private transient Thread installerThread = null;
 
-  private String configName = "default";
-  
+  /**
+   * services which want to know if another service with an interface they are
+   * interested in registers or is released
+   * 
+   * requestor type -> interface -> set of applicable service names
+   */
+  protected final Map<String, Set<String>> interfaceToPossibleServices = new HashMap<>();
+
+  protected final Map<String, Set<String>> typeToNames = new HashMap<>();
+
+  protected final Map<String, Set<String>> interfaceToType = new HashMap<>();
+
+  protected final Map<String, Set<String>> typeToInterface = new HashMap<>();
+
+  /**
+   * FILTERED_INTERFACES are the set of low level interfaces which we are
+   * interested in filtering out if we want to maintain a data structure which
+   * has "interfaces of interest"
+   */
+  protected final static Set<String> FILTERED_INTERFACES = new HashSet<>(Arrays.asList("org.myrobotlab.framework.interfaces.Broadcaster",

```

Comment:
not thrilled about having to maintain this list...  perhaps we can figure a way to clean this up so we don't need to list these interfaces here.